### PR TITLE
JBPM-8165: provide the ability to generate task forms

### PR DESCRIPTION
**Pull request by xiezhang7**
Assembly:
#374 
kiegroup/kie-wb-common#2477

### CODE REVIEW

**Comment by xiezhang7** [comment]
Hi @romartin , I added a method to `ResourceType` to get all the extensions of a resource type for [JBPM-8165](https://issues.jboss.org/browse/JBPM-8165).

**Comment by romartin** [comment]
Hey @xiezhang7 , it looks good to me! Just added a minor review comment, thanks! :+1:

**Comment by xiezhang7** [comment]
Jenkins execute full downstream build

**Comment by xiezhang7** [comment]
Jenkins execute full downstream build

File: kie-api/src/main/java/org/kie/api/io/ResourceType.java [hunk 1]
```diff
@@ -276,6 +280,13 @@ public String getDefaultExtension() {
         return defaultExtension;
     }
 
+    public List<String> getAllExtensions() {
+        final List<String> extensions = new LinkedList<>();
+        extensions.add(defaultExtension);
+        Optional.ofNullable(otherExtensions).map(Arrays::asList).ifPresent(extensions::addAll);
```
[the remark this answers is not in the document]
Comment (xiezhang7) [reply kie-api/src/main/java/org/kie/api/io/ResourceType.java]: Hi @romartin , I have removed `Optional` and added test for `ResourceType` without other extensions. Thanks.

**Comment by romartin** [comment]
@LuboTerifaj I've to merge this as it's part of the assembly with https://github.com/kiegroup/kie-wb-common/pull/2477, which you approved and I just merged... :)

**Pull request merged**